### ISSUE TITLE
SetFullScreen(true) before ShowAndRun fails

### ISSUE THREAD

**Issue by cmitsakis** [fyne-io/fyne#2446]
### Describe the bug:

Calling `w.SetFullScreen(true)` before `w.ShowAndRun()` doesn't work. Program terminates with the the following error:

```
X Error of failed request:  BadValue (integer parameter out of range for operation)
  Major opcode of failed request:  12 (X_ConfigureWindow)
  Value in failed request:  0x0
  Serial number of failed request:  243
  Current serial number in output stream:  249
```

### To Reproduce:

I can reproduce it on i3 and Xfce by running the following code:

```go
package main

import (
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/widget"
)

func main() {
	a := app.New()
	w := a.NewWindow("test")
	w.SetContent(widget.NewLabel("Hello World"))
	w.SetFullScreen(true)
	w.ShowAndRun()
}
```

### Device (please complete the following information):
 - **OS:** Linux
 - **Version:** 5.13.10
 - **Go version:** 1.17
 - **Fyne version:** 2.0.4 and 08dae6d

**Comment by koodimetsa** [comment]
This also happens with Raspberry Pi's latest OS.

**Comment by andydotxyz** [comment]
Fixed on `develop` ready for testing and v2.1
